includes/Core/Assets/Assets.php

```diff
@@ -766,7 +766,19 @@ private function get_inline_user_data() {
 	 * @return array The dashboard sharing inline data to be output.
 	 */
 	private function get_inline_dashboard_sharing_data() {
-		$inline_data = array();
+		$all_roles   = wp_roles()->roles;
+		$inline_data = array( 'roles' => array() );
+
+		foreach ( $all_roles as $role => $details ) {
+			// Filter only the roles with the capability `edit_posts`.
+			$role_to_filter = get_role( $role );
+			if ( $role_to_filter->has_cap( 'edit_posts' ) ) {
```

Comment:
From AC:

> The list of roles in the list should be equal to the set of all WP_Roles on the site with the **edit_post** capability ...

And IB:

> ... and filter only those with the capability **edit_post**

So, you need to check the `edit_post` capability, not the `edit_posts` one.

Reply:
Please refer to the attached screenshot.

<img width="422" alt="Screenshot 2022-01-05 at 8 16 09 PM" src="https://user-images.githubusercontent.com/24408261/148238322-922ac265-1ee1-4347-b2d4-f2547094a2e6.png">

Reply:
@aaemnnosttv the `edit_post` capability expect a post ID to be passed as the second parameter. Can we probably check just `edit_posts` capability to determine whether a role can be sharable or not?

Reply:
@eugene-manuilov yes, this is how it was in the design doc so I must have made a mistake in the AC, but this has been updated now. It should be `edit_posts` 👍

Reply:
Thanks @aaemnnosttv 

@eugene-manuilov I have move it to CR and assigned it back to you :)